src/modules/iasql_functions/iasql.ts

```diff
@@ -923,46 +1109,207 @@ async function getTableMetadata(
 
 /**
  * @internal
- * Return object with formatted value for query and some relevant metadata
+ * Returns the formatted query to be used in the audit log based on the `type` and `inverse` parameters
  */
-function augmentValue(
-  key: string,
-  value: any,
-  metadata: EntityMetadata | RelationMetadata | undefined,
-): { isPrimary: boolean; isAutogenerated: boolean; key: string; formattedValue: string; isJson: boolean } {
-  const augmentedValue = { isPrimary: false, isAutogenerated: false, key, formattedValue: '', isJson: false };
-  let columnMetadata: ColumnMetadata | undefined;
-  // If `metadata instanceof EntityMetadata` means that the `key` is one of the Entity's properties
-  if (metadata && metadata instanceof EntityMetadata) {
-    columnMetadata = metadata.ownColumns.filter(oc => oc.databaseName === key)?.pop();
+async function getFormattedQuery(
+  tableName: string,
+  type: AuditLogChangeType,
+  inverse: boolean,
+  firstEntries: AugmentedValue[],
+  secondEntries?: AugmentedValue[],
+): Promise<string> {
+  let query = '';
+  // Ignore primary keys and autogenerated fields
+  const firstValues = firstEntries.filter(e => (!e.isPrimary && !e.isAutogenerated) || !e.isAutogenerated);
+  const secondValues = secondEntries?.filter(e => (!e.isPrimary && !e.isAutogenerated) || !e.isAutogenerated);
+  // if type is insert or delete and inverse is true, swap types
+  if (inverse && (type === AuditLogChangeType.INSERT || type === AuditLogChangeType.DELETE)) {
+    type = type === AuditLogChangeType.INSERT ? AuditLogChangeType.DELETE : AuditLogChangeType.INSERT;
   }
-  // If `metadata instanceof RelationMetadata` means that the `key` is one of the join columns to another entity.
-  if (metadata instanceof RelationMetadata) {
-    columnMetadata = metadata.joinColumns.filter(jc => jc.databaseName === key)?.pop();
-    if (!columnMetadata) {
-      columnMetadata = metadata.inverseJoinColumns.filter(jc => jc.databaseName === key)?.pop();
+  switch (type) {
+    case AuditLogChangeType.INSERT: {
+      query = format(
+        `
+          INSERT INTO %I (${firstValues.map(_ => '%I').join(', ')})
+          VALUES (${firstValues.map(_ => '%s').join(', ')});
+        `,
+        tableName,
+        ...firstValues.map(e => e.key),
+        ...firstValues.map(e =>
+          e.isJson && Array.isArray(e.value) ? `${formatValue(e)}::jsonb` : formatValue(e),
+        ),
+      );
+      break;
     }
+    case AuditLogChangeType.DELETE: {
+      query = format(
+        `
+          DELETE FROM %I
+          WHERE ${firstValues
+            // We need to add an special case for AMIs since we know the revolve string can be used and it will not match with the actual AMI assigned
+            .filter(e => e.key !== 'ami')
```

Comment:
# :cry:

Reply:
Is there *any* way we could generalize this? There are some other places in the AWS SDK where `resolve:` is allowed that we'll eventually support, and *no one* is going to remember this trickery here, but if we create our own `@awsResolveSupport` decorator or something that can be used here to filter them out we stand a chance of not constantly stubbing our toe on this.